src/wirecloud/commons/static/js/StyledElements/GUIBuilder.js

```diff
@@ -64,6 +55,55 @@
         return new_component;
     };
 
+    extractOptionsFromTextNode = function (element) {
+        var options = {};
+
+        for (var i = 0; i < element.childNodes.length; i++) {
+            var curNode = element.childNodes[i];
+            // find the first non-empty text node
+            if (curNode.nodeType === Node.TEXT_NODE && !(/^\s*$/.test(curNode.nodeValue))) {
+
+                try {
+                    options = JSON.parse(curNode.nodeValue);
+                } catch (e) {
+                    options = {};
+                }
+
+                // remove element to not appear as nested element in result
+                element.removeChild(curNode);
+                break;
+            }
+        }
+
+        return options;
+    };
+
+    processChildren = function processChildren(new_element, builder, element, tcomponents, context) {
```

Comment:
This function can add the children of the template element to the newly created element, and process them accordingly